sdk/core/src/main/java/com/klaviyo/core/utils/ExtTakeIf.kt

```diff
@@ -9,4 +9,5 @@ inline fun <reified T : Any> Any?.takeIf() = let { it as? T }
 /**
  * Extension of takeIf to exclude with a generic type parameter.
  */
-inline fun <reified T : Any> Any?.takeIfNot() = takeIf { it !is T }
+inline fun <SuperType, reified SubType : SuperType> SuperType?.takeIfNot() =
+    takeIf { it !is SubType }
```

Comment:
Previously, within my `takeIfNot` block, we could only know that `it` was `Any`. This extra generic ensures that `it` is part of the desired supertype, but not the excluded subtype.